week4/wordbreak.cc

```diff
@@ -0,0 +1,130 @@
+/* wordbreak.cc
+ * author: connie
+ *
+ * WORD BREAK
+ * Given an input string and a dictionary of words,
+ * find out if the input string can be segmented into a
+ * space-separated sequence of dictionary words.
+ *
+ */
+#include <iostream>
+#include <sstream>
+#include <cassert>
+#include <vector>
+#include <string>
+#include <map>
+
+using namespace std;
+
+//== DECLARATIONS
+bool wordInDict(string& word, vector<string>& dict);
+bool wordBreak_BF(string& s, vector<string>& dict);
+
+//== DEFINITIONS
+//#define DEBUG
+
+//== FUNCTIONS
+/*
+ * bool wordInDict(string& word, vector<string>& dict)
+ * inputs: word - word to check
+           dict - the dictionary (array of strings) with valid words
+ * outputs: True for valid dictionary word, False for invalid word
+ * notes: helper function for iterating thru the array
+ */
+bool wordInDict(string& word, vector<string>& dict)
+{
```

Comment:
It's great that you implemented this in a separated method, stops the code from being crowded and adds modularity :)